### ISSUE TITLE
Hotfixes cannot be "published"

### ISSUE THREAD

**Issue by keithamus** [nvie/gitflow#92]
While release and feature branches can all be published (git flow release/feature publish), hotfix branches cannot.

In our particular development environment, only senior staff can merge to master, meaning that hotfixes need to be pushed to the central repository by junior staff, so a senior member can pull that hotfix branch and then merge to master (git flow hotfix finish). The problem here is that because hotfixes cannot "publish", pushing a hotfix takes on a completely different mantra to pushing a release/feature branch (I'm telling our junior staff that for hotfixes they have to run `git push origin hotfix/1.0.1` as opposed to `git flow hotfix publish`, which they are used to). 

So in summary. Can we get "hotfix publish" added as a command to git flow?

**Comment by cranberyxl** [comment]
Our team could also benefit from this. We work on a web based application and often need to vet hotfixes in a staging environment. While we can fairly painlessly push the hotfix branch to github, the command would be a great convenience.

**Comment by simmerz** [comment]
Additional to this, it would be very useful to have the track function added so that published hotfix branches can easily be brought down. This is something my team could very much do with as well. +1

**Comment by haydn** [comment]
@cranberyxl Exactly the same situation here.

**Comment by maxs** [comment]
Exactly the same situation, please add such feature!

**Comment by mtowers** [comment]
+1

**Comment by sarmbruster** [comment]
+1

**Comment by St0neWan9** [comment]
Really need this +1

**Comment by bartiaco** [comment]
This is definitely needed here.  Love git flow, and I've love to see publishable hotfixes.  We do this for code reviews.

**Comment by phoet** [comment]
+2 :)

**Comment by james-hill** [comment]
+1 Trying to use post-receive hooks + CI server to automatically test hotfixes, then deploy from campfire!

**Comment by haydn** [comment]
Thanks @phoet.

**Comment by mloskot** [comment]
Given the #163 has been merged, why this issue has not been closed?
Is that because `publish hotfix` feature has been merged into develop only and not yet released?

**Comment by zgottlieb** [comment]
hoping this gets released soon! it'll be a very helpful addition.

**Comment by petervanderdoes** [comment]
This is implemented in my fork [git-flow (AVH Edition)](http://bit.ly/S2k1S2)
My fork has diverged to much to make this into an easy pull request for nvie's gitflow,

Checkout the [changelog](http://bit.ly/YiUfPt) for more information about bugfixes and new features implemented in my fork.

**Comment by kretes** [comment]
waiting for that as well

**Comment by poffe** [comment]
+1

**Comment by simobasso** [comment]
+1

**Comment by harmenjanssen** [comment]
Is there any development on this subject? I, too, would really like to have this feature, but the lack of input from the repo's author leads me to wonder wether it'll ever happen?

**Comment by harleensahni** [comment]
+1

**Comment by emanuil-tolev** [comment]
This works for me. Not sure if I misunderstand the issue, but with a relatively recent git flow (installed via apt-get 2 weeks ago on ubuntu 14.04) I can do `git flow hotfix publish test` after `git flow hotfix start test`.

```
Total 0 (delta 0), reused 0 (delta 0)
To <my repo on github>
 * [new branch]      hotfix/test -> hotfix/test
Already on 'hotfix/test'
Your branch is up-to-date with 'origin/hotfix/test'.

Summary of actions:
- A new remote branch 'hotfix/test' was created
- The local branch 'hotfix/test' was configured to track the remote branch
- You are now on branch 'hotfix/test
```

**Comment by simobasso** [comment]
@emanuil-tolev because the git-flow installed via apt it's the git-flow (AVH Edition) a @petervanderdoes 's fork, not the @nvie version.

https://github.com/petervanderdoes/gitflow/wiki

**Comment by gsaslis** [comment]
++ ;

could definitely use this! Thanks! ;)

**Comment by tdakanalis** [comment]
+1. Definitely a good idea!

**Comment by greenman18523** [comment]
++
Would be great to have it!

**Comment by madlew** [comment]
+1 but after 4 years of having this ticket opened it seems there's no hope to have it done

**Comment by emanuil-tolev** [comment]
> +1 but after 4 years of having this ticket opened it seems there's no hope to have it done

Not had any problems with a year and half of https://github.com/petervanderdoes/gitflow , can recommend that fork.

**Comment by gsaslis** [comment]
Ok, thanks @emanuil-tolev - will give that a try!  ; )

**Comment by gsaslis** [comment]
@keithamus does 'closed' mean 'wontfix' ?

**Comment by keithamus** [comment]
Hey @gsaslis; I filed the issue back in 2011. I don't have any ownership of this repo. Closed means I'm no longer interested in this issue personally, and I want to clean up my [issues page](https://github.com/issues) as it is my todo list. If you're interested in this you should maybe file the issue again?

**Comment by gsaslis** [comment]
@keithamus oh, ok, sorry about the confusion there... : ) 
I guess the best solution is still that fork then, since this is now closed

**Comment by keithamus** [comment]
@gsaslis np. I'd say yes - looks like the fork might be the way to go. There's been no activity on this repo for years.

**Comment by jrgtt** [comment]
This is an old thread, but I came here looking for the same solution. This repo is unchanged in 5 years and it looks like it will continue like this, I strongly suggest you to use the [AVH fork](https://github.com/petervanderdoes/gitflow-avh), since it provides more options including `git flow hotfix publish`.


If you installed git-flow using brew, you can simply change it with
```
$ brew uninstall git-flow
$ brew install git-flow-avh
```
